ghost/ghost/src/http/admin/controllers/snippets.controller.ts

```diff
@@ -7,13 +7,25 @@ import {now} from '../../../common/helpers/date.helper';
 import {LocationHeaderInterceptor} from '../../interceptors/location-header.interceptor';
 import {GlobalExceptionFilter} from '../../filters/global-exception.filter';
 import {NotFoundError} from '@tryghost/errors';
+import {Roles} from '../../../common/decorators/permissions.decorator';
+import {PermissionsGuard} from '../../guards/permissions.guard';
+import {AdminAPIAuthentication} from '../../guards/admin-api-authentication.guard';
 
 @Controller('snippets')
+@UseGuards(AdminAPIAuthentication, PermissionsGuard)
 @UseInterceptors(LocationHeaderInterceptor)
 @UseFilters(GlobalExceptionFilter)
 export class SnippetsController {
     constructor(private readonly service: SnippetsService) {}
 
+    @Roles([
```

Comment:
I think even nicer way could be having a controller level `@Roles` decorator and then combine that with method level decorators, to reduce the syntax. But this way is perfectly fine as it's clear straight away "who-can-do-what?"